File: frontend/src/types/workout.ts
import { Exercise } from './exercise';

export interface Workout {
    id: number;
    userId: number;
    date: string;
    duration?: number;
    notes?: string;
    createdAt: string;
    sets: WorkoutSet[];
}

export interface WorkoutSet {
    id: number;
    workoutId: number;
    exerciseId: number;
    exercise?: Exercise;
    setNumber: number;
    weight: number;
    reps: number;
    createdAt: string;
}

export interface WorkoutRequest {
    date: string;
    duration?: number;
    notes?: string;
    sets: WorkoutSetRequest[];
}

export interface WorkoutSetRequest {
    exerciseId: number;
    setNumber: number;
    reps: number;
    weight: number;
}

File: backend/src/main/java/com/treningi/app/entity/Workout.java
package com.treningi.app.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@NoArgsConstructor
@Entity
@Table(name = "workouts")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Workout {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties({"password", "email", "createdAt", "updatedAt", "height", "weight"})
    private User user;

    @Column(nullable = false)
    private LocalDateTime date;

    private Integer duration;

    private String notes;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JsonIgnoreProperties("workout")
    private List<WorkoutSet> sets;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
